set-web/src/main/java/eu/europeana/set/web/service/impl/BaseUserSetServiceImpl.java

```diff
@@ -535,8 +538,34 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet userSetUpdates,
     }
   }
 
+  protected void validateItems(List<String> items) throws ItemValidationException {
+    if(items==null) return;
+    for(String item : items) {
+      if(!item.startsWith(getConfiguration().getItemDataEndpoint())) {
+        throw new ItemValidationException(UserSetI18nConstants.USERSET_ITEM_INVALID_FORMAT, new String[] {item});
+      }
+      else {
+        String withoutBase = item.replace(getConfiguration().getItemDataEndpoint(), "");
+        try {
+          Path pathWithoutBase = Path.of(withoutBase);
+          int nameCount = pathWithoutBase.getNameCount();
+          if(nameCount!=2) {
+            throw new ItemValidationException(UserSetI18nConstants.USERSET_ITEM_INVALID_FORMAT, new String[] {item});
+          }
+          String datasetId = pathWithoutBase.getName(0).toString();
+          if(! datasetId.matches("^[a-z0-9]+$")) {
+            throw new ItemValidationException(UserSetI18nConstants.USERSET_ITEM_INVALID_FORMAT, new String[] {item});
+          }
+        }
+        catch (InvalidPathException ex) {
+          throw new ItemValidationException(UserSetI18nConstants.USERSET_ITEM_INVALID_FORMAT, new String[] {item}, ex);
```

Comment:
please add a comment to indicate that this is implicitly the validation for the recordId part